eng/scripts/CodeChecks.ps1

```diff
@@ -4,7 +4,8 @@
 param (
     [Parameter(Position=0)]
     [string] $ServiceDirectory,
-    [string] $ProjectDirectory
+    [string] $ProjectDirectory,
+    [string] $ExcludeProjects
```

Comment:
Need to define the [Parameter()] attribute  since this is an advanced function (empty parens required in PS). Same below and for any other uses.

Reply:
I don't think it's strictly necessary, as `CmdletBinding` is used.   The [docs](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_functions_advanced_parameters?view=powershell-7.1#parameter-attribute) mention that you need only one or the other to be considered an advanced function.   That said, there's no harm in having both, so I added them in.

Reply:
> Overall good, but with other T1s deprecated (e.g. Key Vault), should we maybe consider a simpler approach (easier to maintain) of moving them out to another directory or something?

That's a wider conversation and beyond the scope of my mandate here.  I would absolutely love to see us split up the service directories, reduce the clutter, and only group together assets that are truly related.   Unfortunately, I've been making that suggestion for quite a while and it hasn't been well-received by the folks that own the guidelines for repository structure.   Please feel free to add your voice to that plea.